### PR TITLE
[sail] Enable static builds and more platforms

### PR DESCRIPTION
**Pull request by HappySeaFox**
**Describe the pull request**

Experimental static build with more platforms.

- What does your PR fix? Fixes #

- Which triplets are supported/not supported? Have you updated the CI baseline?

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes.

### CODE REVIEW

File: ports/sail/portfile.cmake [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
 
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO smoked-herring/sail
-    REF v0.9.0-pre10
-    SHA512 d38a3c7d33495c84d7ada91a4131d32ce61f3fdf32a7e0631b28e7d8492fa4cb672eea09ef8cf0272dabd57f640b090749e3ecd863be577af2b98763873dc57d
-    HEAD_REF master
+    REF 593269c5cd
+    SHA512 fba2c7cde47952a7c0474ef23f1e6f05f5c1f50e533ea6d7e3778126987d114b386ff79000148439c7dc318f8a6133175efbe308ee6ce03fa9d80847bbf78023
+    HEAD_REF feature/static
```
Comment (NancyLi1013) [inline ports/sail/portfile.cmake:8]: Is this a test branch now?

File: ports/sail/portfile.cmake [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
 
+string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" SAIL_STATIC)
+
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
     PREFER_NINJA
     OPTIONS
         -DSAIL_VCPKG_PORT=ON
+        -DSAIL_STATIC=${SAIL_STATIC}
+        -DSAIL_COMBINE_CODECS=ON
```
Comment (NancyLi1013) [inline ports/sail/portfile.cmake:19]: The value for `SAIL_COMBINE_CODECS ` seems to be varied via `SAIL_STATIC `.
https://github.com/smoked-herring/sail/blob/feature/static/CMakeLists.txt#L68

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
The failures on osx:
```
CMake Error at src/sail-codecs-archive/CMakeLists.txt:60 (message):
  Don't know how to build a compiler option for 'AppleClang'
```

File: ports/sail/portfile.cmake (hunk 2 above)
Comment (smoked-herring) [reply ports/sail/portfile.cmake:19]: Yes, but partially. When SAIL_STATIC is ON, the value of SAIL_COMBINE_CODECS is explicitly set to ON. But when SAIL_STATIC  is OFF, user is free to set SAIL_COMBINE_CODECS to ON or OFF. For example, for non-static Windows triplet SAIL_STATIC  is OFF, and SAIL_COMBINE_CODECS  is ON.

File: ports/sail/portfile.cmake (hunk 1 above)
Comment (smoked-herring) [reply ports/sail/portfile.cmake:8]: Yes, it will be merged into master when the PR is ready for review.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
@smoked-herring 

Have you tested the usage of `sail`?

**Comment by smoked-herring** [comment]
@NancyLi1013 Yes. I tested the PR against the demo apps in the repo, everything worked fine.

**Comment by NancyLi1013** [comment]
Thanks for your feedback @smoked-herring.

LGTM now.

**Review by BillyONeal — DISMISSED** [review]
`vcpkg_cmake_wrapper` is the "nuclear option" for cases where the library in question has a find module built into CMake itself; what this one is doing should use an ordinary config file.

Also it should use find_dependency so that `REQUIRED` from callers is correctly respected. ( https://cmake.org/cmake/help/latest/module/CMakeFindDependencyMacro.html )

**Comment by smoked-herring** [comment]
@BillyONeal Thanks! I'll look into this.

**Comment by BillyONeal** [comment]
@smoked-herring You can still fix the bug you're trying to fix there, just with a config instead. Are you still interested in pursuing that or should we merge this as is?

**Comment by smoked-herring** [comment]
@BillyONeal We can merge as is. I've added necessary dependencies into cmake configs in the upstream repo (using find_dependency) and tested them on Windows and Linux.

**Comment by BillyONeal** [comment]
Thanks :)

**Pull request merged**